pkg/validation/validate.go

```diff
@@ -135,11 +149,23 @@ func ValidateLabels(limits LabelValidationLimits, tenantID string, ls []*typesv1
 		if len(l.Value) > limits.MaxLabelValueLength(tenantID) {
 			return NewErrorf(LabelValueTooLong, LabelValueTooLongErrorMsg, phlaremodel.LabelPairsString(ls), l.Value)
 		}
-		var origName string
+		var origName, sanitized string
 		var ok bool
-		if origName, l.Name, ok = SanitizeLabelName(l.Name); !ok {
+		if origName, sanitized, ok = SanitizeLabelName(l.Name); !ok {
 			return NewErrorf(InvalidLabels, InvalidLabelsErrorMsg, phlaremodel.LabelPairsString(ls), "invalid label name '"+origName+"'")
 		}
+
+		// Check if a dup would be created because of sanitization
+		if origName != sanitized {
+			if sanitizedNames[sanitized] {
+				// Add "_dup" suffix to the conflicting sanitized label name
+				sanitized = sanitized + "_dup"
```

Comment:
Nit: Adding `_dup` to the label name, might bring you over `MaxLabelNameLength`, not necessarily a problem.

Reply:
N/A anymore